core/src/main/java/org/elasticsearch/common/io/stream/BytesStreamOutput.java

```diff
@@ -113,18 +113,13 @@ public void flush() throws IOException {
     }
 
     @Override
-    public void seek(long position) throws IOException {
-        if (position > Integer.MAX_VALUE) {
-            throw new IllegalArgumentException("position " + position + " > Integer.MAX_VALUE");
-        }
-
-        count = (int)position;
-        ensureCapacity(count);
+    public void seek(long position) {
+        ensureCapacity(position);
+        count = (int) position;
     }
 
     public void skip(int length) {
-        count += length;
-        ensureCapacity(count);
+        seek((long) count + length);
```

Comment:
My paranoia kicks in here: I don't know order of operations well enough to know if this is `((long) count) + length` or `(long)(count + length)`. Maybe add the extra parens just to make me happy?